Sources/StreamChat/Query/FilterDecoding_Tests.swift

```diff
@@ -0,0 +1,221 @@
+//
+// Copyright © 2021 Stream.io Inc. All rights reserved.
+//
+
+@testable import StreamChat
+import XCTest
+
+class FilterDecoding_Tests: XCTestCase {
```

Comment:
please make sure to test that encoding and decoding are symmetrical so that we dont end up with encoded data failing at decoding

it must not be possible to encode data that cannot be decoded